src/query/models/tags_test.go

```diff
@@ -242,3 +288,133 @@ func TestTagAppend(t *testing.T) {
 
 	assert.Equal(t, expected, tags.Tags)
 }
+
+func TestWriteTagLengthMeta(t *testing.T) {
+	lengths := []int{0, 1, 2, 8, 10, 8, 100, 8, 101, 8, 110, 123456, 12345}
+	l := writer.IntsLength(lengths) + 1 // account for final character
+	require.Equal(t, 42, l)
+	buf := make([]byte, l)
+	count := writeTagLengthMeta(buf, lengths)
+	require.Equal(t, 42, count)
+	assert.Equal(t, []byte("0,1,2,8,10,8,100,8,101,8,110,123456,12345!"), buf)
+}
+
+func buildTags(b *testing.B, count, length int, opts TagOptions, escape bool) Tags {
+	tags := make([]Tag, count)
+	for i := range tags {
+		n := []byte(fmt.Sprint("t", i))
+		v := make([]byte, length)
+		for j := range v {
+			if escape {
+				v[j] = '"'
+			} else {
+				v[j] = 'a'
+			}
+		}
+
+		tags[i] = Tag{Name: n, Value: v}
+	}
+
+	return NewTags(count, opts).AddTags(tags)
+}
+
+var tagBenchmarks = []struct {
+	name                string
+	tagCount, tagLength int
+}{
+	{"10  Tags 10  Length", 10, 10},
+	{"100 Tags 10  Length", 100, 10},
+	{"10  Tags 100 Length", 10, 100},
+	{"100 Tags 100 Length", 100, 100},
+}
+
+const typeQuotedEscaped = IDSchemeType(100)
```

Comment:
What's with the magic number here?

Reply:
Just a big number to use as an "invalid" id scheme type